src/components/AssetHeader/WatchAssetButton.tsx

```diff
@@ -0,0 +1,36 @@
+import type { ButtonProps } from '@chakra-ui/react'
+import { IconButton } from '@chakra-ui/react'
+import type { AssetId } from '@shapeshiftoss/caip'
+import { useCallback } from 'react'
+import { FaRegStar, FaStar } from 'react-icons/fa'
+import { preferences } from 'state/slices/preferencesSlice/preferencesSlice'
+import { selectIsAssetWatched } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+const starEmpty = <FaRegStar />
+const starFilled = <FaStar />
+
+type WatchAssetButtonProps = {
+  assetId: AssetId
+  buttonProps?: ButtonProps
```

Comment:
Seems like there wouldn't be other places we would consume this button and its style would be static, do we want to remove this prop and declare here directly?